lzy/graph-executor-2/src/main/java/ai/lzy/graph/db/impl/GraphDaoImpl.java

```diff
@@ -0,0 +1,155 @@
+package ai.lzy.graph.db.impl;
+
+import ai.lzy.graph.config.ServiceConfig;
+import ai.lzy.graph.db.GraphDao;
+import ai.lzy.graph.model.Graph;
+import ai.lzy.model.db.DbOperation;
+import ai.lzy.model.db.TransactionHandle;
+import jakarta.annotation.Nullable;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+@Singleton
+public class GraphDaoImpl implements GraphDao {
+    private static final Logger LOG = LogManager.getLogger(GraphDaoImpl.class);
+
+    private static final String GRAPH_INSERT_FIELDS_LIST = """
+        id, op_id, status, workflow_id, workflow_name, user_id, graph_description, error_description, 
+        failed_task_id, failed_task_name, last_updated, owner_instance_id""";
+
+    private static final String GRAPH_SELECT_FIELDS_LIST = """
+        id, op_id, status::text as status, workflow_id, workflow_name, user_id, graph_description, 
+        error_description, failed_task_id, failed_task_name, last_updated, owner_instance_id""";
+
+    private final GraphExecutorDataSource storage;
+    private final ServiceConfig config;
+
+    @Inject
+    public GraphDaoImpl(GraphExecutorDataSource storage, ServiceConfig config) {
+        this.storage = storage;
+        this.config = config;
+    }
+
+    @Override
+    public void create(Graph graph, TransactionHandle transaction) throws SQLException {
+        LOG.info("Saving graph: {}", graph);
+
+        DbOperation.execute(transaction, storage, connection -> {
+            try (PreparedStatement st = connection.prepareStatement(
+                "INSERT INTO graph (" + GRAPH_INSERT_FIELDS_LIST + ")"
```

Comment:
it'd be better to use constant string here (not to build new string on each call)